source/slang/hlsl.meta.slang

```diff
@@ -4352,22 +4360,48 @@ __generic<let N : int> float noise(vector<float, N> x)
 /// to this function as necessary in output code, rather than make this
 /// the user's responsibility, so that the default behavior of the language
 /// is more semantically "correct."
-__target_intrinsic(hlsl)
-__target_intrinsic(glsl, nonuniformEXT)
 __glsl_extension(GL_EXT_nonuniform_qualifier)
 [__readNone]
 uint NonUniformResourceIndex(uint index)
 {
-    return index;
+    __target_switch
+    {
+    case hlsl:
+        __intrinsic_asm "NonUniformResourceIndex";
+    case glsl:
+        __intrinsic_asm "nonuniformEXT";
+    case spirv:
+        spirv_asm
+        {
+            OpCapability ShaderNonUniform;
+            OpDecorate $index NonUniform;
```

Comment:
Should this be `[ForceInline]`d to make sure the index we returned is the same SSA value as the one we just decorated?

Reply:
It generates the correct code already, but you're right, forceinline would be more correct.